lingua_franca/format.py

```diff
@@ -483,7 +381,12 @@ def nice_duration(duration, lang=None, speech=True):
     Returns:
         str: timespan as a string
     """
-    if type(duration) is datetime.timedelta:
+    if not lang:
+        lang = get_default_loc()
+    if not is_supported_full_lang(lang):
+        lang = get_full_lang_code(lang)
```

Comment:
Do we need to catch the `UnsupportedLanguageError` if a non-valid lang code is passed? 

Eg `lang='invalid'` from the mycroft-core [test_format.py](https://github.com/MycroftAI/mycroft-core/blob/dev/test/unittests/util/test_format.py#L386)?

Reply:
Yeah, this was briefly discussed in chat but lost in spam and I forgot to document it.

It seems that deprecating `lang=None` was insufficient because of those calls. A straightforward solution would be to deprecate all input *other* than a known lang code, so core ends up printing a warning instead of raising that error.